core/src/services/sftp/pager.rs

```diff
@@ -15,77 +15,76 @@
 // specific language governing permissions and limitations
 // under the License.
 
+use std::pin::Pin;
+
 use async_trait::async_trait;
+use futures::StreamExt;
 use openssh_sftp_client::fs::DirEntry;
+use openssh_sftp_client::fs::ReadDir;
 
 use crate::raw::oio;
 use crate::Result;
 
-pub struct SftpPager {
-    dir: Box<[DirEntry]>,
-    path: String,
-    limit: Option<usize>,
-    complete: bool,
+pub enum SftpPager {
```

Comment:
`oio::Page` has been implemented on all `Option<T: oio::Page>`. So you can use `Option<SftpPager>` directly without wrap again.